src/nodes.coffee

```diff
@@ -4181,9 +4321,21 @@ exports.Param = class Param extends Base
   # `name` is the name of the parameter and `node` is the AST node corresponding
   # to that name.
   eachName: (iterator, name = @name) ->
+    checkAssignabilityOfLiteral = (literal) ->
+      message = isUnassignable literal.value
+      if message
+        literal.error message
+      unless literal.isAssignable()
+        literal.error "'#{literal.value}' can't be assigned"
+
     atParam = (obj, originalObj = null) => iterator "@#{obj.properties[0].name.value}", obj, @, originalObj
+    if name instanceof Call
+      name.error "Function invocation can't be assigned"
```

Comment:
Technically this changes an existing compile-to-JS error message, eg this:
```
({a: b()}) ->
```
used to throw `error: 'b()' can't be assigned` but now throws `error: Function invocation can't be assigned`

Basically during compile-to-JS this error (which is needed to throw an error during AST generation) gets triggered before the existing  error

And the existing error uses eg `name.compile o ` to print `b()` in the error message, but it would be messy to try and have `o` be available here in order to call `name.compile o` (and it seems kind of weird to me to be calling `.compile()` to print out a snippet of (technically transpiled-to-JS) source code in an error message anyway)